### PR TITLE
refactor contact store

### PR DESCRIPTION
**Pull request by umihai29**
## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Additional context

----

> [!IMPORTANT]
> Refactor contact store by introducing `ContactService` for managing GraphQL operations, improving code structure and maintainability.
> 
>   - **Refactoring**:
>     - Introduced `ContactService` to handle GraphQL operations for contacts, replacing inline GraphQL requests in `ContactStore` and `ContactsStore`.
>     - Removed redundant code in `Contact.store.ts` and `Contacts.store.ts` related to direct GraphQL requests.
>   - **GraphQL Operations**:
>     - Added new GraphQL documents for contact operations, including `addContactPhoneNumber`, `addContactSocial`, `addJobRole`, `addTagsToContact`, `archiveContact`, `contactUpdate`, `createContact`, `createContactForOrg`, `deleteContact`, `emailReplace`, `findWorkContactEmail`, `getContact`, `getContacts`, `linkContactWithOrg`, `removeContactPhoneNumber`, `removeTagsFromContact`, `setPrimaryEmailForContact`, `updateContactPhoneNumber`, `updateContactSocial`, `updateJobRole`.
>   - **Service Layer**:
>     - `ContactService` now manages all contact-related GraphQL operations, improving separation of concerns and code organization.
>   - **Miscellaneous**:
>     - Updated `Invoices.service.ts` to fix a variable name inconsistency.
>     - Updated `graphql.ts` to integrate `ContactService` for contact mutations.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=customeros%2Fcustomeros&utm_source=github&utm_medium=referral)<sup> for f43e9c928f15b1600db3d8aa63a06bd51637eed8. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Review by alexopenline — DISMISSED** [review]

File: packages/apps/frontera/src/routes/finder/src/components/Columns/contacts/Cells/email/EmailCell.tsx [hunk 1]
```diff
@@ -213,20 +207,9 @@ export const EmailCell = observer(
...
-                      }
-
-                      return c;
-                    },
-                    { mutate: false },
-                  );
+                  if (idx !== -1) {
+                    contactStore?.value.emails.splice(idx || 0, 1);
+                  }
                   contactStore?.updateEmail(oldEmail || '', idx);
```
Comment (acalinica) [inline packages/apps/frontera/src/routes/finder/src/components/Columns/contacts/Cells/email/EmailCell.tsx]: Should'nt this use contactStore.commit?

File: packages/apps/frontera/src/routes/finder/src/components/Columns/contacts/Cells/jobTitle/JobTitleCell.tsx [hunk 2]
```diff
@@ -1,27 +1,23 @@
...
 }
 
 export const JobTitleCell = observer(({ contactId }: JobTitleCellProps) => {
   const store = useStore();
-  const [isHovered, setIsHovered] = useState(false);
-  const [isEdit, setIsEdit] = useState(false);
-  const jobTitleInputRef = useRef<HTMLInputElement | null>(null);
+  // const [isHovered, setIsHovered] = useState(false);
+  // const [isEdit, setIsEdit] = useState(false);
+  // const jobTitleInputRef = useRef<HTMLInputElement | null>(null);
```
Comment (acalinica) [inline packages/apps/frontera/src/routes/finder/src/components/Columns/contacts/Cells/jobTitle/JobTitleCell.tsx:15]: Are this commented on purpose or you forgot about it?

File: packages/apps/frontera/src/routes/finder/src/components/Columns/contacts/Cells/organization/OrganizationNameCell.tsx [hunk 3]
```diff
@@ -55,28 +57,55 @@ export const OrganizationNameCell = observer(
...
+                    {
+                      metadata: {
+                        id: value.value,
+                        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                      } as any,
+                      id: value.value,
+                      name: value.label,
+                    } as Organization;
+                }
                 contactStore?.linkOrganization(value.value);
```
Comment (acalinica) [inline packages/apps/frontera/src/routes/finder/src/components/Columns/contacts/Cells/organization/OrganizationNameCell.tsx]: Can this be done through ContactsService.mutateOperation?

File: packages/apps/frontera/src/routes/finder/src/components/ContactPreviewCard/ContactPreviewCard.tsx [hunk 4]
```diff
@@ -81,44 +85,39 @@
...
     )?.id;
 
-    if (fromatedUrl === undefined && url.trim() !== '') {
-      contact?.update((contactData) => {
-        const formattedValue =
-          url.includes('https://www') || url.includes('linkedin.com')
-            ? getFormattedLink(url).replace(/^linkedin\.com\//, '')
-            : url;
+    const formattedValue =
+      url.includes('https://www') || url.includes('linkedin.com')
```
[the remark this answers is not in the document]
Comment (acalinica) [reply packages/apps/frontera/src/routes/finder/src/components/ContactPreviewCard/ContactPreviewCard.tsx:96]: Check to see if this suggestion works. It seems better than str.includes().

File: packages/apps/frontera/src/routes/finder/src/components/ContactPreviewCard/ContactPreviewCard.tsx [hunk 5]
```diff
@@ -225,16 +223,30 @@ export const ContactPreviewCard = observer(() => {
...
             onFocus={(e) => e.target.select()}
             className='w-[290px] overflow-hidden text-ellipsis whitespace-nowrap'
+            onBlur={() => {
+              contact.commit();
+              set(
+                contact.value,
+                'latestOrganizationWithJobRole.jobRole.jobTitle',
+                jobRole?.jobTitle,
+              );
+            }}
```
Comment (acalinica) [inline packages/apps/frontera/src/routes/finder/src/components/ContactPreviewCard/ContactPreviewCard.tsx:238]: I'm pretty sure this commit() call before the set() will cause an unintented side-effect for the next commit() call performed on this contact.

Comment (acalinica) [reply packages/apps/frontera/src/routes/finder/src/components/ContactPreviewCard/ContactPreviewCard.tsx:238]: Commit must always be called after you finish mutating the object.

File: packages/apps/frontera/src/routes/finder/src/components/ContactPreviewCard/components/EmailsSection/EmailsSection.tsx [hunk 6]
```diff
@@ -213,14 +208,7 @@ export const EmailsSection = observer(({ contactId }: EmailsSectionProps) => {
...
-                          (c) => {
-                            c.emails.splice(idx, 1);
-
-                            return c;
-                          },
-                          { mutate: false },
-                        );
+                        contactStore?.value.emails.splice(idx, 1);
 
                         contactStore?.updateEmail(email?.email ?? '');
```
Comment (acalinica) [inline packages/apps/frontera/src/routes/finder/src/components/ContactPreviewCard/components/EmailsSection/EmailsSection.tsx]: Should this be contactStore.commit() instead?

**Review by acalinica — APPROVED** [review]

**Pull request merged**